gcc/rust/backend/rust-compile-type.cc

```diff
@@ -148,11 +148,23 @@ TyTyResolveCompile::visit (const TyTy::FnPtr &type)
   tree result_type = TyTyResolveCompile::compile (ctx, type.get_return_type ());
 
   std::vector<tree> parameters;
-  type.iterate_params ([&] (TyTy::BaseType *p) mutable -> bool {
-    tree pty = TyTyResolveCompile::compile (ctx, p);
-    parameters.push_back (pty);
-    return true;
-  });
+
+  // Solution 1
+  // auto &params = type.get_params ();
+  // for (auto &p : params)
+  //   {
+  //     tree pty = TyTyResolveCompile::compile (ctx, p.get_tyty ());
+  //     parameters.push_back (pty);
+  //   }
+
+  // Solution 2
+
+  std::vector<TyTy::BaseType *> p = type.get_params ();
+  for (long unsigned int i = 0; i < p.size (); i++)
+    {
+      tree pty = TyTyResolveCompile::compile (ctx, p[i]);
```

Comment:
```suggestion
  auto params = type.get_params ();
  for (auto &param: params)
    {
      tree pty = TyTyResolveCompile::compile (ctx, param);
```

I think this is more of a style thing, but we have all of these facilities in C++11 already so why not use them :)